### PR TITLE
Adding CS2 egg

### PR DESCRIPTION
**Pull request by Ballaual**
# Description

This PR is a recreation for [#2490](https://github.com/parkervcp/eggs/pull/2490) as i accidentally deleted the branch.

Currently not added but planned for future updates:
Steam Workshop Collections / Maps

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [x] Does your egg use a custom docker image? - ``ghcr.io/parkervcp/steamcmd:sniper``
    * [ ] Have you tried to use a generic image?
    * [x] Did you PR the necessary changes to make it work? -> https://github.com/parkervcp/yolks/pull/220
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
    * [ ] If some script is needed then it is part of a current yolk or a PR to add one
6. [x] The egg was exported from the panel

### CODE REVIEW

File: game_eggs/steamcmd_servers/counter_strike/counter_strike_2/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+For CS2 Valve has decided to merge the game and server files. There will not be standalone dedicated server files.<br>
+
+## Installation/System Requirements
+
+|  | Minimum | Recommended |
+|---------|---------|---------|
+| Processor | minim x86-64-v2 (with popcnt) | --- |
+| RAM | 2 GB | 4 GB or better |
+| Storage | 33 GB | 50 GB (using Workshop Maps etc.) |
+| Packages | Glibc 2.31 | --- |
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/counter_strike/counter_strike_2/README.md:17]: This should not mather for docker aa that is image based

File: game_eggs/steamcmd_servers/counter_strike/counter_strike_2/egg-counter--strike2.json [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    "author": "admin@ballaual.de",
+    "description": "For over two decades, Counter-Strike has offered an elite competitive experience, one shaped by millions of players from across the globe. And now the next chapter in the CS story is about to begin. This is Counter-Strike 2.",
+    "features": [
+        "steam_disk_space"
+    ],
+    "docker_images": {
+        "ghcr.io\/parkervcp\/steamcmd:sniper": "ghcr.io\/parkervcp\/steamcmd:sniper"
+    },
+    "file_denylist": [],
+    "startup": ".\/game\/bin\/linuxsteamrt64\/cs2 -dedicated $( [ \"$VAC_ENABLED\" == \"1\" ] || printf %s ' -insecure' ) -ip 0.0.0.0 -port {{SERVER_PORT}} -tv_port {{TV_PORT}} -maxplayers {{MAX_PLAYERS}} $( [ \"$RCON_ENABLED\" == \"0\" ] || printf %s ' -usercon' ) +net_public_adr {{SERVER_IP}} +game_alias {{GAMEMODE}} +map {{SRCDS_MAP}} +sv_setsteamaccount {{STEAM_GSLT}",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/counter_strike/counter_strike_2/egg-counter--strike2.json:18]: Arround the token a starting " is missing

File: game_eggs/steamcmd_servers/counter_strike/counter_strike_2/egg-counter--strike2.json [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            "field_type": "text"
+        },
+        {
+            "name": "Max Players",
+            "description": "Specifies the maximum amount of players that are able to join the server.",
+            "env_variable": "MAX_PLAYERS",
+            "default_value": "12",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|numeric|min:1|max:64",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/counter_strike/counter_strike_2/egg-counter--strike2.json:70]: You can better hete use between.

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

File: game_eggs/steamcmd_servers/counter_strike/counter_strike_2/egg-counter--strike2.json (hunk 2 above)
Comment (Ballaual) [reply game_eggs/steamcmd_servers/counter_strike/counter_strike_2/egg-counter--strike2.json:18]: thre is no " at all around the token variable as its not needed. The " at the end comes from the startup line itself

File: game_eggs/steamcmd_servers/counter_strike/README.md [hunk 4]
```diff
@@ -4,5 +4,6 @@ Counter-Strike is a series of multiplayer tactical first-person shooter video ga
 
 ## Counter-Strike Titles
 
+- [Counter-Strike 2](counter_strike_2)
```
[the remark this answers is not in the document]
Comment (Ballaual) [reply game_eggs/steamcmd_servers/counter_strike/README.md]: [2adac44](https://github.com/parkervcp/eggs/pull/2725/commits/2adac44e4a532a0bd44789b4644a5188980f8734)

File: game_eggs/steamcmd_servers/counter_strike/counter_strike_2/egg-counter--strike2.json (hunk 3 above)
Comment (Ballaual) [reply game_eggs/steamcmd_servers/counter_strike/counter_strike_2/egg-counter--strike2.json]: [73f47bf](https://github.com/parkervcp/eggs/pull/2725/commits/73f47bf0d2b231f577e34673e6267411706a96c0)

File: game_eggs/steamcmd_servers/counter_strike/counter_strike_2/README.md (hunk 1 above)
Comment (Ballaual) [reply game_eggs/steamcmd_servers/counter_strike/counter_strike_2/README.md]: [acb4bc6](https://github.com/parkervcp/eggs/pull/2725/commits/acb4bc61673feac4520df7043218729b971c9ec7)

**Comment by pedrotski** [comment]
**net_public_adr** doesn't work in CS2 

**game_alias** isn't really something people use as there is no control over it, its game_mode and game_type

**host_workshop_map** and **host_workshop_collection** would be valuable to add (you can only use one or the other) as they must be added as a startup command, unless you exec the server.cfg manually before loading the default map.

In the **server.cfg** you have **sv_parallel_sendsnapshot** set to 1. This isn't a thing in CS2. By default, it's multithreaded, and setting it to 1 will actually make it worse. 

**sv_lan** and **sv_logfile** are already set to 0 so not sure why you're adding and setting them again.

Tbh everything set in server.cfg should be moved to the startup command, so people can overwrite it using their own cfg if they wish. This is one of the biggest complaints/points of confusion we've had was parsing people's server cfgs.

**Comment by Ballaual** [comment]
> **net_public_adr** doesn't work in CS2
> 
> **game_alias** isn't really something people use as there is no control over it, its game_mode and game_type
> 
> **host_workshop_map** and **host_workshop_collection** would be valuable to add (you can only use one or the other) as they must be added as a startup command, unless you exec the server.cfg manually before loading the default map.
> 
> In the **server.cfg** you have **sv_parallel_sendsnapshot** set to 1. This isn't a thing in CS2. By default, it's multithreaded, and setting it to 1 will actually make it worse.
> 
> **sv_lan** and **sv_logfile** are already set to 0 so not sure why you're adding and setting them again.
> 
> Hibernate when empty should be removed from a default config. The description is incorrect for the current setting too.
> 
> Tbh everything set in server.cfg should be moved to the startup command, so people can overwrite it using their own cfg if they wish. This is one of the biggest complaints/points of confusion we've had was parsing people's server cfgs.

see commit [bc81c2d](https://github.com/parkervcp/eggs/pull/2725/commits/bc81c2d9635bd0a4091363dfee7b722339dd18f8)
Exept the game alias part i 100% agreee. It should be userfriendly in my opinion and let the user set game_mode and game_type by himself isnt userfriendly as there are only these 4 specified gamemodes yet. and the aliases works well and are correct according to the documentation.

the workshop part is definetly a thing that could be added in a later version. i am not familiar with it and also do not have the time at the moment to properly testing this out.

**Comment by pedrotski** [comment]
> > **net_public_adr** doesn't work in CS2
> > **game_alias** isn't really something people use as there is no control over it, its game_mode and game_type
> > **host_workshop_map** and **host_workshop_collection** would be valuable to add (you can only use one or the other) as they must be added as a startup command, unless you exec the server.cfg manually before loading the default map.
> > In the **server.cfg** you have **sv_parallel_sendsnapshot** set to 1. This isn't a thing in CS2. By default, it's multithreaded, and setting it to 1 will actually make it worse.
> > **sv_lan** and **sv_logfile** are already set to 0 so not sure why you're adding and setting them again.
> > Hibernate when empty should be removed from a default config. The description is incorrect for the current setting too.
> > Tbh everything set in server.cfg should be moved to the startup command, so people can overwrite it using their own cfg if they wish. This is one of the biggest complaints/points of confusion we've had was parsing people's server cfgs.
> 
> see commit [bc81c2d](https://github.com/parkervcp/eggs/pull/2725/commits/bc81c2d9635bd0a4091363dfee7b722339dd18f8) Exept the game alias part i 100% agreee. It should be userfriendly in my opinion and let the user set game_mode and game_type by himself isnt userfriendly as there are only these 4 specified gamemodes yet. and the aliases works well and are correct according to the documentation.
> 
> the workshop part is definetly a thing that could be added in a later version. i am not familiar with it and also do not have the time at the moment to properly testing this out.

There is currently 10 loadable game modes for CS2.

- Casual
- Co-op Strike
- Competitive
- Custom
- Deathmatch
- Demolition
- Guardian
- Training Day
- Training
- Weapons Expert
- Wingman

Workshop is easy to add too as they can simply remain blank, and they will be unused. For each they will use a workshop id such as 3070557958

**Comment by Ballaual** [comment]
> There is currently 10 loadable game modes for CS2.
> 
> * Casual
> * Co-op Strike
> * Competitive
> * Custom
> * Deathmatch
> * Demolition
> * Guardian
> * Training Day
> * Training
> * Weapons Expert
> * Wingman

source?

**Comment by pedrotski** [comment]
> > There is currently 10 loadable game modes for CS2.
> > 
> > * Casual
> > * Co-op Strike
> > * Competitive
> > * Custom
> > * Deathmatch
> > * Demolition
> > * Guardian
> > * Training Day
> > * Training
> > * Weapons Expert
> > * Wingman
> 
> source?

I'm the source. I ran one of the most popular CSGO and CS2 servers in the world, and we host CS2 servers for lots of people.

**Comment by QuintenQVD0** [comment]
@parkervcp This looks good to me? Can this be merged or would you like to do some extra tests on it?

**Review by parkervcp — APPROVED** [review]

**Comment by pedrotski** [comment]
I would highly suggest changing the game mode alias to game type and mode to be future-proof. It's already out of date again, since CS2 has now added Arms Race as well.

**Comment by QuintenQVD0** [comment]
> I would highly suggest changing the game mode alias to game type and mode to be future-proof. It's already out of date again, since CS2 has now added Arms Race as well.

What? Can you be a bit more specific what you want to be changed as I did not get that. It did work the egg so ?

**Comment by Ballaual** [comment]
> > I would highly suggest changing the game mode alias to game type and mode to be future-proof. It's already out of date again, since CS2 has now added Arms Race as well.
> 
> What? Can you be a bit more specific what you want to be changed as I did not get that. It did work the egg so ?

He wants the game_alias cvar beeing removed and replaced by its separate functions game_mode and game_type. If done this way the Readme needs an update too explaining the respective values and what the user need to add, as these settings are fir advanced users only.

The recent added gamemode armsrace needs to be added to the GAMEMODE variable in this egg version, thats true. I did not had the time to take a look at the latest CS2 update last week..

https://developer.valvesoftware.com/wiki/Counter-Strike_2/Dedicated_Servers

https://developer.valvesoftware.com/wiki/Counter-Strike:_Global_Offensive/Game_Modes

**Comment by QuintenQVD0** [comment]
> > > I would highly suggest changing the game mode alias to game type and mode to be future-proof. It's already out of date again, since CS2 has now added Arms Race as well.
> > 
> > 
> > What? Can you be a bit more specific what you want to be changed as I did not get that. It did work the egg so ?
> 
> He wants the game_alias cvar beeing removed and replaced by its separate functions game_mode and game_type. If done this way the Readme needs an update too explaining the respective values and what the user need to add, as these settings are for advanced users only. In my opinion this can be customized by each Ptero host himself if needed.
> 
> The recent added gamemode armsrace needs to be added to the GAMEMODE variable in this egg version, thats true. I did not had the time to take a look at the latest CS2 update last week..
> 
> However which way should be implemented to this rgg i will either add the new gamemode or chamge the whole startup logic this evening. Waiting for Parkers and your response to this topic.
> 
> https://developer.valvesoftware.com/wiki/Counter-Strike_2/Dedicated_Servers
> 
> https://developer.valvesoftware.com/wiki/Counter-Strike:_Global_Offensive/Game_Modes

To be honnest. I do not play this game. Do what yiu all think is the best and just works.

**Comment by Ballaual** [comment]
To clarify:
game_alias competitive = game_type 0 game_mode 1
game_alias wingman = game_type 0 game_mode 2

For casual users = game_alias is the best choice.
In the end you can use either the game_alias or the other method.
Both will work.

**Comment by pedrotski** [comment]
> To clarify: game_alias competitive = game_type 0 game_mode 1 game_alias wingman = game_type 0 game_mode 2
> 
> For casual users = game_alias is the best choice. In the end you can use either the game_alias or the other method. Both will work.

It's not going to be the best choice because it means you constantly need to monitor and update the egg for every game mode that is released. Using type and mode won't have these issues, and the egg will require less maintenance from end users and maintainers alike.

**Comment by Ballaual** [comment]
> Even in the official docs they don't use game_alias on the game mode page.

For CS2 the official docs they only shows the game_alias.
https://developer.valvesoftware.com/wiki/Counter-Strike_2/Dedicated_Servers

However i just uploaded an updated version... and kindly ask you to test this out and confirm as i do not have the time right now to test this as needed... I am not at home this week!

**Comment by Ballaual** [comment]
I guess the map variable needs an update too with the new armsrace maps? or better said.. the predefined inputs should also be removed (to keep maintenance as low as possible).

**Comment by pedrotski** [comment]
> For CS2 the official docs they only shows the game_alias. Dont know which other official docs u are reading. https://developer.valvesoftware.com/wiki/Counter-Strike_2/Dedicated_Servers

Game mode page. And yes do the same for map otherwise it will simply lead to people coming to Discord asking how to update it once an update comes out. 

I understand what you're doing, trying to offer an easier experience, but it's going to be harder for end users to self serve once issues arise.

**Comment by Ballaual** [comment]
> > For CS2 the official docs they only shows the game_alias. Dont know which other official docs u are reading. https://developer.valvesoftware.com/wiki/Counter-Strike_2/Dedicated_Servers
> 
> Game mode page. And yes do the same for map otherwise it will simply lead to people coming to Discord asking how to update it once an update comes out.
> 
> I understand what you're doing, trying to offer an easier experience, but it's going to be harder for end users to self serve once issues arise.

has been updated.

**Comment by Ballaual** [comment]
From my side this has been tested and confirmed working.

**Comment by QuintenQVD0** [comment]
@pedrotski You have anything to say as it is also fine for me.

**Review by QuintenQVD0 — APPROVED** [review]
LGTM!

**Pull request merged**